layers/newrelic/Dockerfile

```diff
@@ -0,0 +1,43 @@
+ARG PHP_VERSION
+ARG NEW_RELIC_LICENSE_KEY
+ARG NEW_RELIC_DAEMON_ADDRESS
+FROM bref/build-php-$PHP_VERSION AS ext
+
+# Build the New Relic Agent install silently for PHP Agent 9.14
+# Then append php.ini with key information including a license key and 
+# location of the daemon coming from an ENV VAR for Lambda
+
+RUN \
+  curl -L http://download.newrelic.com/php_agent/release/newrelic-php5-9.14.0.290-linux.tar.gz | tar -C /tmp -zx && \
+  export NR_INSTALL_USE_CP_NOT_LN=1 && \
+  export NR_INSTALL_SILENT=1 && \
+  /tmp/newrelic-php5-*/newrelic-install install && \
+  rm -rf /tmp/newrelic-php5-* /tmp/nrinstall* 
+  
+RUN echo $' \n \
+extension=newrelic.so \n \
+newrelic.appname = ${NEW_RELIC_APP_NAME} \n \
+newrelic.license = ${NEW_RELIC_LICENSE_KEY} \n \
+newrelic.daemon.address = ${NEW_RELIC_DAEMON_ADDRESS} \n \
+newrelic.logfile = ${NEW_RELIC_LOG_FILE} \n \
+newrelic.loglevel = ${NEW_RELIC_LOG_LEVEL} \n \
+newrelic.daemon.app_connect_timeout = ${NEW_RELIC_DAEMON_APP_CONNECT_TIMEOUT} \n \
+newrelic.daemon.start_timeout = ${NEW_RELIC_DAEMON_START_TIMEOUT} \n \
+' >> /opt/bref/etc/php/php.ini
```

Comment:
These are things specific to the application. It is the user that should add these. Right?

I suggest removing these.

Reply:
To work, the agent must be capable of connecting to a remote running daemon. 
so 
extension=newrelic.so (needed to tell PHP to load the newrelic.so file)
newrelic.appname = ${NEW_RELIC_APP_NAME}.   (Needs to be set or the Lambda will default report to PHP Application)
newrelic.license = ${NEW_RELIC_LICENSE_KEY}   (Needed to report to your account)
newrelic.daemon.address = ${NEW_RELIC_DAEMON_ADDRESS}  (Needed to tell the agent where the remote daemon is located)

The following configuration settings are definitely options but having control of them via the Lambda Environment variables is much more scalable and you can then have your settings like key and daemon address stores in something like KMS if you wanted. For example, using Serverless we defined ENV VARS and those ENV VARS become the settings below. In particular it's useful to set log file to STDERR location so it will log to Cloudwatch.

newrelic.logfile = ${NEW_RELIC_LOG_FILE} \n \
newrelic.loglevel = ${NEW_RELIC_LOG_LEVEL} \n \
newrelic.daemon.app_connect_timeout = ${NEW_RELIC_DAEMON_APP_CONNECT_TIMEOUT} \n \
newrelic.daemon.start_timeout = ${NEW_RELIC_DAEMON_START_TIMEOUT} \n \

I can remove them, but I'm interested in where in the Lambda you would propose the end user sets these values. PHP will only read the php.ini, a file in the Scan Directory or .user.ini in the local folder. Maybe I'm oblivious to something obvious but I only believe I can amend php.ini or anything in the scan dir. So unless I use .user.ini is your suggestion that the user of this extension should put it in their PHP code using **ini_set** ?

Reply:
Yes, I suggest that we dont add any config here. We should let the default values provided by newrelic take effect. 

In your application you should add `php/conf.d/mysettings.ini` where you write: 

```
newrelic.appname = ${NEW_RELIC_APP_NAME}
newrelic.license = ${NEW_RELIC_LICENSE_KEY}
newrelic.daemon.address = ${NEW_RELIC_DAEMON_ADDRESS}
newrelic.logfile = ${NEW_RELIC_LOG_FILE}
newrelic.loglevel = ${NEW_RELIC_LOG_LEVEL}
newrelic.daemon.app_connect_timeout = ${NEW_RELIC_DAEMON_APP_CONNECT_TIMEOUT}
newrelic.daemon.start_timeout = ${NEW_RELIC_DAEMON_START_TIMEOUT} 
```

Another user may specify hard coded values here instead of the environment variables.

Reply:
We can make that change, the agent won't work at all without a license key and remote daemon_address though so anyone who implements it will need to know to add a mysettings.ini file. 

If we remove these settings, then we can remove the line copying the php.ini to the extension layer. The only change from the core php.ini to the one in my layer is these config changes. 

So I can make the container literally just install New Relic and copy over the .so file to where it needs to be. Then add additional documentation on implementation of configuration.